ddtrace/api.py

```diff
@@ -73,7 +75,13 @@ def send_services(self, services):
         log.debug("reported %d services", len(services))
         return response
 
-    def _put(self, endpoint, data):
+    def _put(self, endpoint, data, count=0):
         conn = httplib.HTTPConnection(self.hostname, self.port)
-        conn.request("PUT", endpoint, data, self._headers)
+
+        headers = self._headers
+        if count:
+            headers = copy.copy(self._headers)
```

Comment:
here it's better to not use a `copy.copy`. In general in Python when handling dictionaries it's enough to use `dict(self._headers)` to create a new shallow copy. Because here your intention is only to add a new field and **never** to alter already assigned fields, a shallow copy using the `dict` keyword is faster and better.

Also the field is "private" so we assume nobody will change the content of the shallow copy other than in this function.

```python
a = {'foo': 42, 'bar': 'fish'}

In [2]: %timeit copy.copy(a)
1000000 loops, best of 3: 695 ns per loop
In [3]: %timeit dict(a)
1000000 loops, best of 3: 272 ns per loop
```